From 09d4fe2eff468cb5f725d8c561b23eef86338c74 Mon Sep 17 00:00:00 2001
From: Yuval Caduri <cyuval@marvell.com>
Date: Thu, 17 Aug 2017 16:27:06 +0300
Subject: [PATCH 1936/2241] net: mvpp2x: Simplify mac_address update

Existing mv_pp2x_set_mac_address fn stopped and started
the netdevice during mac_address update.
This is not required by HW, and therefore unnecessary.
This update removes that code.

Change-Id: I8491ff4278b0d19a145a06c0e1f5ebe4420834d2
Signed-off-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43181
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 22 ++--------------------
 1 file changed, 2 insertions(+), 20 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index b28ce56..9cdb670 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -4183,7 +4183,6 @@ static void mv_pp2x_set_rx_mode(struct net_device *dev)
 
 static int mv_pp2x_set_mac_address(struct net_device *dev, void *p)
 {
-	struct mv_pp2x_port *port = netdev_priv(dev);
 	const struct sockaddr *addr = p;
 	int err;
 
@@ -4192,28 +4191,11 @@ static int mv_pp2x_set_mac_address(struct net_device *dev, void *p)
 		goto error;
 	}
 
-	if (!netif_running(dev)) {
-		err = mv_pp2x_prs_update_mac_da(dev, addr->sa_data);
-		if (!err)
-			return 0;
-		/* Reconfigure parser to accept the original MAC address */
-		err = mv_pp2x_prs_update_mac_da(dev, dev->dev_addr);
-		goto error;
-	}
-
-	mv_pp2x_stop_dev(port);
-
 	err = mv_pp2x_prs_update_mac_da(dev, addr->sa_data);
 	if (!err)
-		goto out_start;
-
-	/* Reconfigure parser accept the original MAC address */
+		return 0;
+	/* Reconfigure parser to accept the original MAC address */
 	err = mv_pp2x_prs_update_mac_da(dev, dev->dev_addr);
-	if (err)
-		goto error;
-out_start:
-	mv_pp2x_start_dev(port);
-	return 0;
 
 error:
 	netdev_err(dev, "fail to change MAC address\n");
-- 
2.7.4

